### PR TITLE
PYIC-1710: Add a statemachine config yaml file per env

### PR DESCRIPTION
**Pull request by dannyhayes-gds**
### What changed
Add a new State machine config yaml file per environment.

With this change I was also able to remove the use of the CRIResponses and just use JourneyResponses to control the cri/start calls. This is temporary until we move all of this into step functions.

### Why did it change
Gives us the ability to easily update the journey on a per environment basis.

### Issue tracking

- [PYIC-1710](https://govukverify.atlassian.net/browse/PYIC-1710)

### CODE REVIEW

File: lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/StateMachineInitializer.java [hunk 1]
```diff
@@ -33,4 +43,41 @@ public Map<String, State> initialize() throws IOException {
...
+                                            "statemachine/integration-statemachine-config.yaml"))
+                            .getFile());
+        } else {
+            return new File(
+                    Objects.requireNonNull(
+                                    classLoader.getResource(
+                                            "statemachine/production-statemachine-config.yaml"))
+                            .getFile());
+        }
+    }
```
Comment (kerrr) [inline lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/statemachine/StateMachineInitializer.java:57]: Can we avoid all the duplicate file loading code by constructing the file name first? We could also make sure the env portion of the file name matches the env var, special case for dev excepted.

File: lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/ProcessJourneyStepHandler.java [hunk 2]
```diff
@@ -59,8 +59,8 @@ public ProcessJourneyStepHandler(
 
     @ExcludeFromGeneratedCoverageReport
     public ProcessJourneyStepHandler() throws IOException {
-        this.stateMachine = new StateMachine(new StateMachineInitializer());
         this.configurationService = new ConfigurationService();
+        this.stateMachine = new StateMachine(new StateMachineInitializer(configurationService));
```
Comment (kerrr) [inline lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/ProcessJourneyStepHandler.java]: Does passing in the whole config service help here? Can we just pass in the env var?

**Review by kerrr — DISMISSED** [review]
Couple of minor nits.

File: lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/ProcessJourneyStepHandler.java (hunk 2 above)
Comment (dannyhayes-gds) [reply lambdas/process-journey-step/src/main/java/uk/gov/di/ipv/core/processjourneystep/ProcessJourneyStepHandler.java]: Ah yeah good shout, I'll change this just to pass in the env var instead

**Review by kerrr — APPROVED** [review]

**Pull request merged**